aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/TestLocalAliyunOSS.java

```diff
@@ -0,0 +1,184 @@
+
+    byte[] bytes = new byte[2000];
+    random.nextBytes(bytes);
+
+    oss.putObject(bucketName, "object1", wrap(bytes));
+
+    oss.putObject(bucketName, "object2", wrap(bytes));
+
+    assertThrows(() -> oss.deleteBucket(bucketName), OSSErrorCode.BUCKET_NOT_EMPTY);
+
+    oss.deleteObject(bucketName, "object1");
+    assertThrows(() -> oss.deleteBucket(bucketName), OSSErrorCode.BUCKET_NOT_EMPTY);
+
+    oss.deleteObject(bucketName, "object2");
+    oss.deleteBucket(bucketName);
+    Assert.assertFalse(doesBucketExist(bucketName));
+
+    oss.createBucket(bucketName);
+  }
+
+  @Test
+  public void testPutObject() throws IOException {
+    byte[] bytes = new byte[4 * 1024];
+    random.nextBytes(bytes);
+
+    String bucketNotExist = String.format("bucket-not-existing-%s", UUID.randomUUID());
+    assertThrows(() -> oss.putObject(bucketNotExist, "object", wrap(bytes)), OSSErrorCode.NO_SUCH_BUCKET);
+
+    PutObjectResult result = oss.putObject(bucketName, "object", wrap(bytes));
+    Assert.assertEquals(AliyunOSSMockLocalStore.md5sum(wrap(bytes)), result.getETag());
+  }
+
+  @Test
+  public void testDoesObjectExist() {
+    Assert.assertFalse(oss.doesObjectExist(bucketName, "key"));
+
+    Assert.assertFalse(oss.doesObjectExist(bucketName, "key"));
+
+    byte[] bytes = new byte[4 * 1024];
+    random.nextBytes(bytes);
+    oss.putObject(bucketName, "key", wrap(bytes));
+
+    Assert.assertTrue(oss.doesObjectExist(bucketName, "key"));
+    oss.deleteObject(bucketName, "key");
+  }
+
+  @Test
+  public void testGetObject() throws IOException {
+    String bucketNotExist = String.format("bucket-not-existing-%s", UUID.randomUUID());
+    assertThrows(() -> oss.getObject(bucketNotExist, "key"), OSSErrorCode.NO_SUCH_BUCKET);
+
+    assertThrows(() -> oss.getObject(bucketName, "key"), OSSErrorCode.NO_SUCH_KEY);
+
+    byte[] bytes = new byte[2000];
+    random.nextBytes(bytes);
+
+    oss.putObject(bucketName, "key", new ByteArrayInputStream(bytes));
+
+    byte[] actual = new byte[2000];
+    IOUtils.readFully(oss.getObject(bucketName, "key").getObjectContent(), actual);
+
+    Assert.assertArrayEquals(bytes, actual);
+    oss.deleteObject(bucketName, "key");
+  }
+
+  @Test
+  public void testGetObjectWithRange() throws IOException {
+
+    byte[] bytes = new byte[2000];
+    random.nextBytes(bytes);
+    oss.putObject(bucketName, "key", new ByteArrayInputStream(bytes));
+
+    byte[] actual = new byte[2000];
+    int start = 0;
+    int end = 1999;
+    GetObjectRequest getObjectRequest = new GetObjectRequest(bucketName, "key");
+    getObjectRequest.setRange(start, end);
+    IOUtils.readFully(oss.getObject(getObjectRequest).getObjectContent(), actual);
+
+    Assert.assertArrayEquals(bytes, actual);
+    oss.deleteObject(bucketName, "key");
+  }
+
+  private InputStream wrap(byte[] data) {
+    return new ByteArrayInputStream(data);
+  }
+
+  private boolean doesBucketExist(String bucket) {
+    try {
+      oss.createBucket(bucket);
```

Comment:
Em.. seems we don't provide any bucket existence checking API in the local OSS.  I forgot why we don't implement it ,  seems it's fussy to mock all the object metadata..

Reply:
it is the simplest way to simulate this api behavior.  we can mock further version if necessary later per upper layer need. may keep it as it is here.